File: HelperJobby/DTOs/Job/CreateIncompleteJobDTO.cs
using ApplicationDomain.Enums;

namespace HelperJobby.DTOs.Job;

public class CreateIncompleteJobDTO
{
    public string JobTitle { get; set; }
    public int NumberOfOpenings { get; set; }
    public string Language { get; set; }
    public string Location { get; set; }
    public List<JobTypes> JobType { get; set; }
    public CreateUpdateSalaryDTO? Salary { get; set; }
    public List<Schedules> Schedule { get; set; }
    public List<EmployeeBenefits> Benefits { get; set; }
    public string? ContactEmail { get; set; }
    public bool? ResumeRequired { get; set; }
    public string? Description { get; set; }
}

File: HelperJobby/client-app/src/DTOs/jobRelatetedDTOs/CreateIncompleteJobDTO.ts
import JobTypes from "../../enums/modelDataEnums/JobTypes";
import Schedules from "../../enums/modelDataEnums/Schedules";
import EmployeeBenefits from "../../enums/modelDataEnums/EmployeeBenefits";
import {CreateUpdateSalaryDTO} from "./CreateUpdateSalaryDTO";
import {JobLocationTypes} from "../../enums/modelDataEnums/JobLocationTypes";
import {ResumeRequirementOptions} from "../../enums/modelDataEnums/ResumeRequirementOptions";

export interface CreateIncompleteJobDTO {
    jobTitle: string;
    numberOfOpenings: number;
    language: string;
    location: string;
    locationCountry : string;
    jobLocationType : JobLocationTypes;
    jobType?: JobTypes[];
    salary? : CreateUpdateSalaryDTO;
    schedule?: Schedules[];
    benefits?: EmployeeBenefits[];
    contactEmail?: string;
    contactPhoneNumber? : string;
    resumeRequired?: ResumeRequirementOptions;
    description?: string;
}